conans/model/build_info.py

```diff
@@ -485,10 +521,13 @@ class DepsCppInfo(_BaseDepsCppInfo):
     def __init__(self):
         super(DepsCppInfo, self).__init__()
         self._dependencies = OrderedDict()
-        self.configs = {}
+        self._configs = {}
 
     def __getattr__(self, config):
-        return self.configs.setdefault(config, _BaseDepsCppInfo())
+        return self._configs.setdefault(config, _BaseDepsCppInfo())
+
+    def get_configs(self):
```

Comment:
I am curious why this is not just a property?

Reply:
Probably there was no reason for it. During the refactor I renamed attribute `configs` to private `_configs` and added a function to get the configs (so I can put a breakpoint there and see who/when/why it was called)... and btw, we are creating `__len__` config 😅 (change will arrive in a different PR).

I usually write functions first and then properties... do you prefer a property here?

Reply:
Yes, I would say it is a bit more idiomatic, plus it will not break some potential user that was writing a custom generator and iterating over the configs too.